Usage analytics

etdsoft opened the pull request:
### Summary

Today we don't have a great way to understand how users are navigating Dradis, whether they're getting stuck, or what features are used (or ignored). It makes it tricky to allocate time & resources to the right tasks.

After much searching it seems that the open-source privacy-friendly analytics from Plausible - https://github.com/plausible/analytics strikes the right balance of giving us a hint about usage, without compromising privacy.

This PR adds the analytics opt in/out step to the wizard to allow users to choose to sare telemetry info.


### Check List

- [x] Added a CHANGELOG entry
- [x] Commit message has a detailed description of what changed and why.
- [x] Add a [CE Privacy FAQ](https://dradis.com/ce/privacy.html) page to the site.

etdsoft committed:
SessionsController - inherit from SetupRequired

Under some conditions (i.e. session timeout while your DB is wiped, in
dev or sandbox environment), you end up in /login with an instance that
hasn't been set up properly yet.

etdsoft committed:
Add analytics partial and include in layouts

etdsoft committed:
Merge branch 'release-4.19.0' into analytics

etdsoft committed:
Wizard - new step for privacy-friendly analytics

Add open source, privacy-friendly analytics from Plausible to help us
understand better how people are using Dradis.

Don't opt people in by default.

etdsoft committed:
CHANGELOG - add analytics to setup wizard

etdsoft committed:
Merge branch 'develop' into analytics

etdsoft committed:
Configuration - lint, remove extra line

MattBudz committed:
save users a click + tweak copy

etdsoft committed:
Setup::Passwords#create - update next step

aapomm commented on app/controllers/setup/analytics_controller.rb:
```diff
@@ -0,0 +1,22 @@
+module Setup
+  class AnalyticsController < BaseController
+    def new
+    end
```
```suggestion
    def new; end

```

aapomm commented on app/views/layouts/application.html.erb:
```diff
@@ -10,6 +10,7 @@
     <%= javascript_importmap_tags %>
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <%= render 'layouts/analytics' %>
```
The name of the file is a bit misleading as it isn't a layout. What about moving this to `shared` and going with a more obvious naming: `shared/_plausible_tracking` ?

MattBudz replied:
What about `layouts/head/_plausible_tracking` or just`layouts/head/_plausible`?

etdsoft replied:
> The name of the file is a bit misleading as it isn't a layout

The convention is that layout partials live under layouts/

I'm happy with what it is tbh.

Could be persuaded to use `layouts/shared/` but having a folder for a single file seems to be overcomplicating things. Similar to `layouts/head/`

etdsoft committed:
Merge branch 'develop' into analytics

etdsoft committed:
Update app/controllers/setup/analytics_controller.rb

Co-authored-by: Aaron Manaloto <mail@aapomm.io>

etdsoft committed:
Spec - add fixtures to skip the Setup wizard

etdsoft committed:
AnalyticsController - fix typo

etdsoft committed:
AnalyticsController - lint

etdsoft committed:
ControllerMacros - lint

etdsoft committed:
Revert SessionsController parent change

This breaked specs that assume that unauthenticated access would result
in a redirect to /login. When Sessions inherits from Setup, and if we
haven't done the Setup process in the specs, instead of /login we end up
in /setup/*

etdsoft committed:
Merge branch 'develop' into analytics

etdsoft committed:
SessionsController should inherit from SetupRequired

This is to ensure you're not attempting to log in without having run
through the Setup Wizard yet. It's a rare situation that can happen
either in Development or if in the Sandbox environment your DB gets
nuked while you were signed in. Then your session expires, and you're
sent to /login on a pristine DB, which before this PR the controller
wouldn't detect.

After this change, even in that scenario, /login redirects to /setup if
the DB becomes pristine.

The problem is that our Specs were relying on "login trickery" to bypass
the Setup Wizard rather than addressing the problem head on. This PR
also creates the necessary Fixtures to effectively bypass the Wizard for
all tests.

etdsoft committed:
Spec - move Setup Wizard mocking to support_helper

etdsoft committed:
Setup::Kits - lint

etdsoft committed:
Setup::Progress partial - tweak for Pro parity

etdsoft committed:
Adjust Sessions spec

etdsoft committed:
SetupRequired - we needed a new step

etdsoft committed:
Sessions spec - lint

etdsoft committed:
User Searches spec - parity

etdsoft committed:
Update spec/support/support_helper.rb

Co-authored-by: Aaron Manaloto <mail@aapomm.io>

etdsoft committed:
Apply suggestions from code review

Co-authored-by: Aaron Manaloto <mail@aapomm.io>

etdsoft committed:
Merge pull request #1468 from dradis/rspec/fix-welcome

SessionsController should inherit from SetupRequired

etdsoft committed:
Merge branch 'develop' into analytics

etdsoft committed:
Analytics - conditional render

Also, rename to make vendor explicit.

etdsoft committed:
Setup::Analytics - refine description, link to FAQ

caitmich committed:
use array for cache key
